compiler/src/iree/compiler/Codegen/Passes.td

```diff
@@ -474,6 +474,12 @@ def ConvertToNVVM :
   let constructor = "mlir::iree_compiler::createConvertToNVVMPass()";
 }
 
+def LLVMGPULayoutAnalysisAndDistribution :
```

Comment:
Can we make it a transform dialect op instead of a pass? That might be easier to push some early execution test to validate that things are correct.
This would work the same as a pass as it would run on the whole function.

Reply:
Sounds good. Will change.